pkg/ccl/changefeedccl/changefeed_stmt.go

```diff
@@ -675,6 +667,18 @@ func getTargetsAndTables(
 	targets := make([]jobspb.ChangefeedTargetSpecification, len(rawTargets))
 	seen := make(map[jobspb.ChangefeedTargetSpecification]tree.ChangefeedTarget)
 
+	hasControlChangefeed, err := p.HasRoleOption(ctx, roleoption.CONTROLCHANGEFEED)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	var requiredPrivilegePerTable privilege.Kind
+	if hasControlChangefeed {
+		requiredPrivilegePerTable = privilege.SELECT
+	} else {
+		requiredPrivilegePerTable = privilege.CHANGEFEED
```

Comment:
If we have both the roleoption and `privilege.Changefeed`, shouldn't this be valid?

Reply:
I agree this is a bit weird and I do call it out in the release note. If I remember right we decided it this way so that if you want, you can keep the legacy behavior where `REVOKE SELECT ON TABLE` prevents a highly-privileged user from getting the data using select or changefeeds.